include/Popup.hpp

```diff
@@ -14,15 +14,14 @@ struct PopupListeners
 
 class Popup : public PopupListeners
 {
-  public:
-    Popup(View *child, wlr_surface *surface);
-    ~Popup();
+public:
+  Popup(wlr_surface *surface);
+  ~Popup();
 
-    wlr_surface *surface;
-
-  private:
-    View *child;
-
-    void handle_new_popup(wl_listener *listener, void *data);
-    void handle_destroy_popup(wl_listener *listener, void *data);
+  wlr_surface *surface;
+  std::unique_ptr<Popup> popup;
```

Comment:
Is this in the case where a popup can have its own popups ?

Reply:
Yes. I deduced that should be the case since handle_new_popup exists.